src/main/java/io/javalin/http/JavalinServlet.kt

```diff
@@ -80,15 +81,7 @@ class JavalinServlet(val config: JavalinConfig) : HttpServlet() {
                     return // don't write body
                 }
             }
-            if (gzipShouldBeDone(ctx)) {
-                GZIPOutputStream(res.outputStream, true).use { gzippedStream ->
-                    res.setHeader(Header.CONTENT_ENCODING, "gzip")
-                    resultStream.copyTo(gzippedStream)
-                }
-                resultStream.close()
-                return
-            }
-            resultStream.copyTo(res.outputStream) // no gzip
+            dynamicCompressionHandler.compressResponse(res)
```

Comment:
All compression-invoking code has been moved into the DynamicCompressionHandler object